nucypher/blockchain/eth/actors.py

```diff
@@ -969,25 +1029,50 @@ def initialize_stake(self,
                          fund_now: bool = True
                          ) -> Stake:
 
-        if password and not checksum_address:
-            staker = self.__create_staker(password=password)
+        # Existing Staker address
+        if checksum_address:
+            if not is_checksum_address(checksum_address):
+                raise ValueError(f"{checksum_address} is an invalid EIP-55 checksum address.")
+
+            try:
+                staker = self.__stakers[checksum_address]
+            except KeyError:
+                if checksum_address not in self.__accounts:
+                    raise ValueError(f"{checksum_address} is an unknown wallet address.")
+                else:
+                    staker = Staker(is_me=True, checksum_address=checksum_address, blockchain=self.blockchain)
+
+        # New Staker address (no address passed)
         else:
-            staker = Staker(is_me=True, checksum_address=checksum_address, blockchain=self.blockchain)
+            if not password:
+                raise ValueError(f'No checksum address or password supplied to initialize new stake.')
+            staker = self.__create_staker(password=password)
```

Comment:
I need to dive in w/ a debugger, but I'm imagining that a random wallet is generated here?  Maybe that needs to be called out in at least a comment on this layer?

Reply:
I agree - more comments are needed here.  This area needs some thinking, ( see related Issue #1128 and #1129 ).  In the case of a software wallet - the RPC endpoint for account creation is used.  In the cased of a hardware wallet, an unused account is derived and used (which is always the last index geth loads for hardware wallets).

Reply:
Yeah, so this is a bit of a lava bridge.  I think it's a little nerve-racking for `self.__create_staker` to, without affirmative direction to do so, use a heretofore unused checksum address.

Reply:
From a brid's-eye view does it make sense to implement the feature at all? (#1129 ).  What's the user's experience like if they need to create accounts to stake with?

Reply:
@jMyles  - Also, If you read above, the user is prompted for consent to use a new account.